example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,606 @@
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderRightActions?: (
+    progressAnimatedValue: SharedValue,
+    dragAnimatedValue: SharedValue,
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+  ) => React.ReactNode;
+
+  useNativeAnimations?: boolean;
+
+  animationOptions?: Record<string, unknown>;
+
+  /**
+   * Style object for the container (`Animated.View`), for example to override
+   * `overflow: 'hidden'`.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+}
+
+export interface ExposedFunctions {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  reset: () => void;
+}
+
+const Swipeable = forwardRef<
+  SwipeableProps & React.RefAttributes<ExposedFunctions>
+>(function Swipeable(
+  props: SwipeableProps,
+  ref: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+) {
+  const [rowState, setRowState] = useState<number>(0);
+
+  const dragX = useSharedValue<number>(0);
+  const rowTranslation = useSharedValue<number>(0);
+  const leftWidth = useSharedValue<number>(0);
+  const rightOffset = useSharedValue<number>(0);
+  const rowWidth = useSharedValue<number>(0);
+
+  // todo: if things don't work, set all the default values from the original file
+  // they are set everywhere thoughout this file, many functions set the default states themselves
+
+  const defaultProps = {
+    friction: 1,
+    overshootFriction: 1,
+    useNativeAnimations: true,
+  };
+
+  useEffect(() => {
+    if (!props.friction) props.friction = defaultProps.friction;
+    if (!props.overshootFriction)
+      props.overshootFriction = defaultProps.overshootFriction;
+    if (!props.useNativeAnimations)
+      props.useNativeAnimations = defaultProps.useNativeAnimations;
+
+    updateAnimatedEvent();
+  }, []);
+
+  /* todo: export to the exposed store
+  
+  const shouldComponentUpdate = (props: SwipeableProps) => {
+    if (
+      props.friction !== props.friction ||
+      props.overshootLeft !== props.overshootLeft ||
+      props.overshootRight !== props.overshootRight ||
+      props.overshootFriction !== props.overshootFriction
+    ) {
+      updateAnimatedEvent(props);
+      return true;
+    }
+    return false;
+  };
+  */
+
+  const transX = useSharedValue(0); // only IV
+  const showLeftAction = useSharedValue(0); // can AV
+  const leftActionTranslate = useSharedValue(0); // only IV
+  const showRightAction = useSharedValue(0); // can AV;
+  const rightActionTranslate = useSharedValue(0); // only IV
```

Comment:
Please don't use acronyms in the comments.

Reply:
removed those all together, was only there to keep track of what happened in the old API